backend/src/app/http/oauth.clj

```diff
@@ -81,14 +81,28 @@
                           :timeout 6000
                           :method :get}))
 
-          (validate-response [{:keys [status body] :as res}]
-            (when-not (= 200 status)
+          (retrieve-emails []
+            (if (some? (:emails-uri provider))
+              (http-client {:uri (:emails-uri provider)
+                            :headers {"Authorization" (str (:type tdata) " " (:token tdata))}
+                            :timeout 6000
+                            :method :get})
+              (p/resolved {:status 200})))
+
+          (validate-response [[retrieve-res emails-res]]
+            (when-not (= 200 (:status retrieve-res))
```

Comment:
I think it's better to check the status to be in a range. 20X codes are all "OK" i think it's better to be safe and check if it's in the range [200, 300)